cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/remote/RemoteProcessController.java

```diff
@@ -29,15 +31,27 @@ public interface RemoteProcessController {
    */
   boolean isRunning() throws Exception;
 
+  /**
+   * Returns the runtime status {@link RuntimeJobStatus} of the remote process for the program execution.
+   *
+   * @throws Exception if not able to determine the status of the remote process
+   */
+  RuntimeJobStatus getStatus() throws Exception;
+
   /**
    * Graceful shutdown of the remote process
    * @throws Exception if not able to terminate the remote process
    */
   void terminate() throws Exception;
 
+  /** @deprecated */
+  @Deprecated
+  void kill() throws Exception;
```

Comment:
this one can be removed, since this interface is an internal one (not in api or spi module)